### PR TITLE
Retract pre-module versions and v1.5.2

### PR DESCRIPTION
**Pull request by liggitt**
Proposed as part of https://github.com/kubernetes/kubernetes/issues/98267

Based on v1.5.1 content, intended to merge to the release-1.5 branch and have a v1.5.2 tag cut

### CODE REVIEW

**Comment by liggitt** [comment]
cc @sttts @deads2k @lavalamp

**Comment by liggitt** [comment]
/cc @sttts @dims 

where is the pull-client-go-build job configured? would like to understand the failure there

**Comment by liggitt** [comment]
It looks like go1.15 will not build a main module that declares it wants go1.16 and uses go1.16 retract directives, but works fine referencing a module like that as a dependency. Since k8s.io/client-go is not a main module, this will not cause problems for anyone actually using/building v1.5.1 with an older go version (which I expect to be vanishingly small anyway).

**Comment by liggitt** [comment]
/hold cancel

This is ready for review. Once merged, we will manually tag the resulting commit on the release-1.5 branch as v1.5.2

**Comment by dims** [comment]
/approve

will leave `/lgtm` to @sttts

**Comment by liggitt** [comment]
ha... apparently we need approval from https://github.com/kubernetes/client-go/blob/release-1.5/OWNERS

**Comment by liggitt** [comment]
looks like sttts can merge/tag manually

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+retract (
+	// pre-module versions are obsolete
+	v12.0.0+incompatible
+	// pre-module versions are obsolete
+	v11.0.0+incompatible
+	// pre-module versions are obsolete
+	v10.0.0+incompatible
+	// pre-module versions are obsolete
+	v9.0.0+incompatible
```
Comment (nikhita) [inline go.mod:13]: We also have a `v9.0.0-invalid` - https://github.com/kubernetes/client-go/tree/v9.0.0-invalid

Should that be included here?

Comment (liggitt) [reply go.mod:13]: excellent catch... I don't know how I missed that

**Comment by sttts** [comment]
In go-build job:
```
go: errors parsing go.mod:
/home/prow/go/src/k8s.io/client-go/go.mod:5: unknown block type: retract
```

**Comment by sttts** [comment]
Otherwise LGTM.

**Comment by nikhita** [comment]
> In go-build job:

@sttts that's expected because the job is still on go1.15 - https://github.com/kubernetes/test-infra/pull/21003#discussion_r582126020

The job is also meant to run against master, and k/k is still on go1.15

**Comment by liggitt** [comment]
See https://github.com/kubernetes/client-go/pull/932#issuecomment-785217673 for evaluation of the CI build error

**Comment by nikhita** [comment]
I think override needs to be explicitly enabled..

Merging manually.

**Pull request merged**